### ISSUE TITLE
Philips/Signify Hue Wall Switch Module (model 929003017102, RDM004) with direct binding => no dimming

### ISSUE THREAD

**Issue by SkyhawkXava** [Koenkk/zigbee2mqtt#25242]
### What happened?

Zigbee2MQTT version: 1.42.0-2 (via Home Assistant with SkyConnect)
Hello,

I have several Signify Hue wall switch modules (model 929003017102, RDM004) directly bound to various light bulbs. Turning the lights on and off works so far, but the light bulbs (IKEA/Tradfri and Müller Licht/Tint) cannot be dimmed. Of course, the switch's "device mode" option has been set to "single_push_button".

The Hue Wall Switch Module is directly bound to the “LevelCtrl” and “OnOff” clusters for the respective Zigbee groups.

Under “Directly control Zigbee groups” for Philips 929003017102 in the Z2M device database, it states that dimming should be possible with direct binding if the “LevelCtrl” cluster is bound to the respective group.

### What did you expect to happen?

Dimming Lights via ZigBee-Group by using Hue Wall Switch Module (long press/hold the push button).

### How to reproduce it (minimal and precise)

1.) Create a group of lights/bulbs.
2.) Pair a "Hue Wall Switch Module" to z2m.
3.) Set option "device mode" of the "Hue Wall Switch Module" to "single_push_button".
4.) Bind Cluster "groups" to Coordinator.
5.) Add the "Hue Wall Switch Module" to the above group.
6.) Unbind Clusters "LevelCtrl", "OnOff" and "manuSpecificPhilips" to Coordinator.
7.) Bind Clusters "LevelCtrl" and "OnOff" to above group.
8.) Push and hold the push button, which ist connected to the "Hue Wall Switch Module".

### Zigbee2MQTT version

1.42.0

### Adapter firmware version

7.2.1.0 build 144

### Adapter

EZSP v10

### Setup

Add-on on Home Assistant OS

**Comment by S-blok** [comment]
Same problem over here. 
Using the Hue wall switch module via SLZB-06M and z2m. Followed the steps on https://www.zigbee2mqtt.io/devices/929003017102.html#philips-929003017102. The on/off works, the dimming won't work.

When I hold down the push button, the lamp turns on or off. It doesn't dim

**Comment by jzernisch** [comment]
Same here.

**Comment by tsloms** [comment]
> Same problem over here. Using the Hue wall switch module via SLZB-06M and z2m. Followed the steps on https://www.zigbee2mqtt.io/devices/929003017102.html#philips-929003017102. The on/off works, the dimming won't work.
> 
> When I hold down the push button, the lamp turns on or off. It doesn't dim

Same here. Exactly same layout. 
Seems that the switch activates as soon as the button is closed (rather than released). Is there a way to fix that?

**Comment by philipgo** [comment]
According to the Zigbee2MQTT docs this should work. Can anyome confirm it worked in the past?

**Comment by tsloms** [comment]
> According to the Zigbee2MQTT docs this should work. Can anyome confirm it worked in the past?
No, I can't confirm that, as I only recently tried it out.

**Comment by SkyhawkXava** [comment]
On the one hand, the hardware revisions RDM001 and RDM004 behave differently. However, this also seems to depend heavily on the installed firmware version. Even with RDM001 running the latest firmware version, there still appear to be issues.

See also issue #19941 (Can´t add Hue wall switch to Zigbee group).

**Comment by philipgo** [comment]
Mine is RDM004 and dimming does not work. With RDM001 it appears to be working only with older firmware. 

Can the hardware version be found on the device without pairing it?

**Comment by tsloms** [comment]
I've got the RDM004 version. Firmware version 2.60.16. Model says 929003017102.

**Comment by philipgo** [comment]
I bought a used RDM001 (model 9290030171). Unfortunately it came with firmware 2.77.35 and shows the same problems.

**Comment by janhenk102** [comment]
Issue apparently not resolved yet. Today I migrated my Hue Wall Switch to Z2M and I wanted it to bind it to a group. Switching on/off works fine, but dimming (long press) doesn’t. I use RDM001 and a Smlight SLZB-06M.

On/off toggle is on press, it doesn’t “wait” for de long press.

**Comment by tsloms** [comment]
Is there any update on this, whether and how this can possibly be fixed?

**Comment by niklasschloegel** [comment]
I have the same issue, but adding the wall switch module to the group does not work for me.

**Comment by trollix** [comment]
I have the same issue, but adding the wall switch module to the group does not work for me.

**Comment by tkoenig** [comment]
Just bought the same device because I read the instructions on https://www.zigbee2mqtt.io/devices/929003017102.html and wanted this functionality. (alternately dimming with single rocker).

On off works, but no dimming :( 
Is there anything I can do to help debugging this?

**Comment by ehallq** [comment]
I would also like to know if anyone else how found a solution to this. I have 2 old wall switches where the binding works as it should, but I need 4 more... Haven't found any other product that is able to do this. (EU)
Binding with 1 single rocker that also alternate the dimming. This allows full control of any light even if coordinator is down. Kind of the closest you can get to a hardwired light, but with a smartbulb that can do so many other things. The current behavior of the updated hue wall switches is unfortunately useless since if the bulb is dimmed very low, and coordinator is down, you can't increase the brightness. 

Tried a bunch of "smart  zigbee buttons" that I had planned to "hack" to connect to my old switches by connecting two wires to it, but nothing behaves like the old Hue Wall Switches.

An alternative is finding a module that can be connected with no neutral, that also can have decoupled mode. But there are not such alternatives in Europe...

**Comment by SashaBusinaro** [comment]
+1 same problem

**Comment by andiwirs** [comment]
same problem with my Smlight SLZB-06

**Comment by andrei-lazarov** [comment]
I see v2.77.35 is available for both models in Z2M. 
I'll look for older versions, so you guys could try downgrading.

---

@ehallq An alternative is **AVATTO modules + custom firmware**: [romasku/tuya-zigbee-switch](https://github.com/romasku/tuya-zigbee-switch)
_(You can install it wirelessly like an OTA update)_
It currently supports mains-powered switches, binding onOff + levelCtrl, detached mode etc.
Battery-powered remotes/switches will be supported in the future.

No-Neutral switches need a load to function. Minimum value depends on hardware and firmware.
Smart bulbs draw less than 1W when brightness=0%, which is too low!
This is why you can't find No-Neutral + decoupled mode.

The custom firmware allows this, but you'll need a dummy load (capacitor), or at least one gang in attached mode with a dumb bulb.

Also SONOFF ZBMINIR2 supports binding onOff on original firmware. But no levelCtrl and it needs Neutral.

**Comment by gonzague** [comment]
ohhh my @andrei-lazarov thats super interesting! exactly what I was trying to build with an ESP32 :D

**Comment by philipgo** [comment]
Still a significant problem, please do not close.

**Comment by andrei-lazarov** [comment]
I checked the history of zigbee-ota. Did not find older versions.

Maybe it has a different combination of binding/unbinding?
For example some IKEA remotes need binding genBasic to a group, and unbinding everything else (the coordinator)

Someone could try the Hue hub to confirm if dimming still works, and sniff Zigbee traffic to see what they do differently.

Should we add a warning to the Z2M doc "update v2.77.35 may remove dimming"? Or should we retract the update?

**Comment by mundschenk-at** [comment]
Not stale

**Comment by MartB** [comment]
Hmm seems like something philips broke i guess? Or we are configuring it wrong somehow? For me 2.60.16 already exhibits that behavior on my RDM004.

**Comment by mundschenk-at** [comment]
Yes, RDM004 never had a firmware not showing this behavior. It's a deliberate change by Signify (making their various battery-powered remotes work the same way).

**Comment by jzernisch** [comment]
problem persists
